### PR TITLE
Fixes formatting in wsg and adds a code example guideline

### PR DESCRIPTION
**Pull request by dipikabh**
#### Summary

This PR is essentially to update the writing 'code example' guidelines in our writing style guide. Added the following as per @estelle's suggestion:
"In general, the code example should not only demonstrate the syntax of the feature and how it is used, but also highlight the purpose and situations in which a web developer might want or need to use the feature."

This PR also fixes some of the following issues in the writing style guide:
- Removes the unnecessary indent characters `>`.
- Changes some H4s to bullet points because we're recommending in the style guide to not go lower than three heading levels.
- Changes some H4s to H3s by removing the parent H3.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [X] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/mdn/writing_guidelines/writing_style_guide/index.md [hunk 1]
```diff
@@ -462,8 +432,8 @@ Keep the following dos and don'ts in mind while creating headings for subsection
 - **Don't use styles and classes within headings.** This includes the {{HTMLElement("code")}} element for code terms.
   So don't format the heading as "Using the `SuperAmazingThing` interface".
   It should instead just be "Using the SuperAmazingThing interface".
-- **Avoid using macros within headings.** (except for certain macros that are specifically designed to be used in headings)
-- **Don't create "bumping heads".** These are headings followed immediately by a subheading, with no content text in between.
+- **Avoid using macros within headings.** (Except for certain macros that are specifically designed to be used in headings).
```
Comment (teoli2003) [inline files/en-us/mdn/writing_guidelines/writing_style_guide/index.md:435]: We don't have such macro anymore.

```suggestion
- **Avoid using macros within headings.** Ever.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
A nit and this is good to go.

File: files/en-us/mdn/writing_guidelines/writing_style_guide/index.md (hunk 1 above)
Comment (dipikabh) [reply files/en-us/mdn/writing_guidelines/writing_style_guide/index.md:435]: I combined this with the previous point as "**Don't use styles, classes, or macros within headings.**

**Comment by dipikabh** [comment]
Thanks @teoli2003 for reviewing these changes 🙏!

**Review by teoli2003 — APPROVED** [review]
👍 (You are lucky I saw it today, it is a bank holiday here :-) )

**Pull request merged**